src/allotropy/parsers/lines_reader.py

```diff
@@ -53,28 +53,34 @@ def pop(self) -> Optional[str]:
         return line
 
     def pop_if_match(self, match_pat: str) -> Optional[str]:
-        if self.match(match_pat):
-            return self.pop()
-        return None
+        return self.pop() if self.match(match_pat) else None
 
     def pop_data(self) -> Optional[str]:
         self.drop_empty()
         return self.pop()
 
-    def drop_until(self, match_pat: str) -> Optional[str]:
+    def drop_until(
+        self,
+        match_pat: str,
+        inclusive: bool = False,  # noqa: FBT001, FBT002
```

Comment:
Actually don't want to ignore these QA checks, which generally guide use to avoid adding bool flags to functions. 

This is an anti-pattern to avoid because:

1. It can be hard to tell what a short-named bool flag does for future engineers
2. It can lead to functions with several bool flags changing the behavior of the function, making it hard to tell from the call what a function is doing.

The general good pattern here is to create two functions, one for each behavior, e.g. in this case you could make the function:

```
def drop_up_to(self, match_pat: str):
    self.drop_until(match_pat)
    self.pop()
```

The good pattern is to create two functions in this case, e.g. in this case you could split into